masonry/src/passes/update.rs

```diff
@@ -471,6 +474,12 @@ pub(crate) fn run_update_focus_pass(root: &mut RenderRoot) {
         // We also request accessibility, because build_access_node() depends on the focus state.
         if let Some(prev_focused) = prev_focused {
             if root.widget_arena.has(prev_focused) {
+                if was_ime_active {
+                    // IME was active, but the next focused widget is going to receive the
+                    // Ime::Disabled event. Synthesize an Ime::Disabled event here and send it to
+                    // the widget about to be unfocused.
+                    run_on_text_event_pass(root, &TextEvent::Ime(winit::event::Ime::Disabled));
+                }
```

Comment:
One edge case here is that this causes inconsistencies if `set_focus` ends up being called in this event.
At the very least, we need to move the setting of `self.global_state.focused_widget` on line 504 to use `next_focused` instead.

Reply:
Ah, of course. I tried to be careful, e.g. with the position where the event is triggered, but missed the `EventCtx::set_focus` method.